src/cargo/util/errors.rs

```diff
@@ -86,7 +86,8 @@ impl CargoError {
             &CargoErrorKind::Parse(_) |
             &CargoErrorKind::Git(_) |
             &CargoErrorKind::Internal(_) |
-            &CargoErrorKind::CargoTestErrorKind(_) => false
+            &CargoErrorKind::CargoTestErrorKind(_) |
+            &CargoErrorKind::__Nonexhaustive { .. } => false
```

Comment:
This seems like something is wrong somewhere...

Should we ask that error chain has an option not to generate Nonexhaustive?

Error chain issue: https://github.com/rust-lang-nursery/error-chain/issues/182

Reply:
Is this wrong though? It seems like it's fine to match "nonexhaustive errors" inside of Cargo itself?

Reply:
Hm, yeah, kinda buy "all consumers must add `_`, but we can use __Nonexaustive".